packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -329,19 +329,15 @@ export default class Proxy {
 
   async callProxyMethod(id, method, params, expectStream = false) {
     const proxy = await this._getProxy(id)
-    if (proxy.address === undefined) {
-      if (proxy.vmUuid === undefined) {
-        throw new Error(
-          'proxy VM and proxy address should not be both undefined'
-        )
-      }
+    if (proxy.address === undefined && proxy.vmUuid === undefined) {
+      throw new Error('proxy VM and proxy address should not be both undefined')
+    }
 
+    if (proxy.vmUuid !== undefined) {
       const vm = this._app.getXapi(proxy.vmUuid).getObjectByUuid(proxy.vmUuid)
-      if (
-        (proxy.address = extractIpFromVmNetworks(
-          vm.$guest_metrics?.networks
-        )) === undefined
-      ) {
+      proxy.address =
+        extractIpFromVmNetworks(vm.$guest_metrics?.networks) ?? proxy.address
```

Comment:
I'm not sure about the `?? proxy.address`. IMO, if the proxy has a VM, we should 100% use its information (ie leave it as "broken" if it doesn't have an IP). And if we want to use a custom IP, we should first unbind/delete the VM from the proxy.

What do you think?

Reply:
I don't know why we should unbind/delete the VM from the proxy given that the custom IP will refers to its IP.

Reply:
I agree, we should never use `proxy.address` if attached to a VM.

Reply:
Why? What is the scenario which will use the proxy address?